x-pack/legacy/plugins/rollup/public/crud_app/sections/job_create/steps_config/index.js

```diff
@@ -63,83 +88,81 @@ export const stepIdToStepConfigMap = {
         rollupCron,
         rollupPageSize,
         rollupDelay,
+        clonedRollupId,
       } = fields;
-
-      const errors = {
-        id: validateId(id),
+      return {
+        id: validateId(id, clonedRollupId),
         indexPattern: validateIndexPattern(indexPattern, rollupIndex),
         rollupIndex: validateRollupIndex(rollupIndex, indexPattern),
         rollupCron: validateRollupCron(rollupCron),
         rollupPageSize: validateRollupPageSize(rollupPageSize),
         rollupDelay: validateRollupDelay(rollupDelay),
       };
-
-      return errors;
     },
   },
   [STEP_DATE_HISTOGRAM]: {
-    defaultFields: {
-      dateHistogramField: null,
-      dateHistogramInterval: null,
-      dateHistogramTimeZone: 'UTC',
+    getDefaultFields: (overrides = {}) => {
+      const defaults = {
+        dateHistogramField: null,
+        dateHistogramInterval: null,
+        dateHistogramTimeZone: 'UTC',
+      };
+
+      return {
+        ...defaults,
+        ...pick(overrides, Object.keys(defaults)),
+      };
     },
     fieldsValidator: fields => {
-      const {
-        dateHistogramField,
-        dateHistogramInterval,
-      } = fields;
+      const { dateHistogramField, dateHistogramInterval } = fields;
 
-      const errors = {
+      return {
         dateHistogramField: validateDateHistogramField(dateHistogramField),
         dateHistogramInterval: validateDateHistogramInterval(dateHistogramInterval),
       };
-
-      return errors;
     },
   },
   [STEP_TERMS]: {
-    defaultFields: {
-      terms: [],
+    getDefaultFields: (overrides = {}) => {
+      return {
+        terms: [],
+        ...pick(overrides, ['terms']),
+      };
     },
   },
   [STEP_HISTOGRAM]: {
-    defaultFields: {
-      histogram: [],
-      histogramInterval: undefined,
+    getDefaultFields: overrides => {
+      return {
+        histogram: [],
+        histogramInterval: undefined,
+        ...pick(overrides, ['histogram', 'histogramInterval']),
+      };
     },
     fieldsValidator: fields => {
-      const {
-        histogram,
-        histogramInterval,
-      } = fields;
+      const { histogram, histogramInterval } = fields;
 
-      const errors = {
+      return {
         histogramInterval: validateHistogramInterval(histogram, histogramInterval),
       };
-
-      return errors;
     },
   },
   [STEP_METRICS]: {
-    defaultFields: {
-      metrics: [],
+    getDefaultFields: (overrides = {}) => {
+      return {
+        metrics: [],
+        ...pick(overrides, ['metrics']),
+      };
     },
     fieldsValidator: fields => {
-      const {
-        metrics,
-      } = fields;
+      const { metrics } = fields;
 
-      const errors = {
+      return {
         metrics: validateMetrics(metrics),
       };
-
-      return errors;
     },
   },
   [STEP_REVIEW]: {
-    defaultFields: {
-      startJobAfterCreation: false
-    }
+    getDefaultFields: () => ({}),
```

Comment:
just want to confirm here that `startJobAfterCreation` is no longer needed?

Reply:
Yep - this is an artefact of an earlier, abandoned approach.